rackcity/models/network_port.py

```diff
@@ -65,6 +74,12 @@ def delete_network_connection(self):
     class Meta:
         constraints = [
             models.UniqueConstraint(
-                fields=['asset', 'port_name'],
-                name='unique network port names on assets'),
+                fields=["asset", "port_name"],
+                name="unique network port names on assets",
+            ),
         ]
+
+    def save(self, *args, **kwargs):
+        if self.mac_address is not None:
+            self.mac_address = format_mac_address(self.mac_address)
+        super(NetworkPort, self).save(*args, **kwargs)
```

Comment:
I approve if you confirm that the front end will handle character validation (ie everything between delimiters is hex 0-9, a-f)

Reply:
yes, the backend should only every recieve mac addresses that are 6 byte hex strings with no delimiters, :, or - delimiters